pkgs/by-name/sn/snowflake-cli/package.nix

```
@@ -0,0 +1,86 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  installShellFiles,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "snowflake-cli";
+  version = "3.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "snowflakedb";
+    repo = "snowflake-cli";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-QBYdzKKkakgwCD12bN7CerjfvMnxvt2cJ0NqTIjHkWw=";
+  };
+
+  build-system =
+    (with python3Packages; [
+      hatch-vcs
+      hatchling
+      pip
+    ])
+    ++ [ installShellFiles ];
+
+  dependencies = with python3Packages; [
+    jinja2
+    pluggy
+    pyyaml
+    rich
+    requests
+    requirements-parser
+    setuptools
+    tomlkit
+    typer
+    urllib3
+    gitpython
+    pydantic
+    snowflake-connector-python
+  ];
+
+  nativeCheckInputs = with python3Packages; [
+    pytestCheckHook
+    syrupy
+    coverage
+    pytest-randomly
```

```suggestion
    pytest-randomly
    pytest-xdist
```
To support parallel tests.